### PR TITLE
Print stack trace in assertNotError

### LINKED ISSUE #462: Print error stacktrace when TestObserver.assertNotError() fails
Printing `"Source failed"` is not very useful when `test().assertNotError()` is used in a test. What do you think of printing the error stacktrace as well?

### PR DESCRIPTION
**Pull request by arkivanov**
Fixes #462

### CODE REVIEW

File: reaktive-testing/src/commonMain/kotlin/com/badoo/reaktive/test/base/TestObserverExt.kt [hunk 1]
```diff
@@ -1,27 +1,50 @@
...
+import kotlin.test.fail
 
 fun TestObserver.assertError() {
     assertTrue(isError, "Source did not fail")
 }
 
 fun TestObserver.assertError(expectedError: Throwable) {
-    assertEquals(expectedError, error, "Source error does not match")
+    val error = error
+    if (error == null) {
```
Comment (CherryPerry) [inline reaktive-testing/src/commonMain/kotlin/com/badoo/reaktive/test/base/TestObserverExt.kt:17]: You can beautify it a little. But up to you, `when` can be overkill here and I don't know, will it be optimised by compiler.

```kotlin
when (val error = error) {
     null -> fail()
     else -> { ... }
}
```

Comment (arkivanov) [reply reaktive-testing/src/commonMain/kotlin/com/badoo/reaktive/test/base/TestObserverExt.kt:17]: Personally I don't like replacing `if-else` with `when` :-)

Comment (arkivanov) [reply reaktive-testing/src/commonMain/kotlin/com/badoo/reaktive/test/base/TestObserverExt.kt:17]: Personally I don't like replacing `if-else` with `when`, let's keep it like this?

Comment (CherryPerry) [reply reaktive-testing/src/commonMain/kotlin/com/badoo/reaktive/test/base/TestObserverExt.kt:17]: Keep it.

**Pull request merged**